proposals/user-defined-compound-assignment.md

```diff
@@ -0,0 +1,568 @@
+an `override` modifier can be used for this purpose.
+
+ECMA-335 already "reserved" the following special names for user defined increment operators:
+| Name | Operator |
+| -----| -------- |
+|op_AdditionAssignment|'+=' |
+|op_SubtractionAssignment|'-=' |
+|op_MultiplicationAssignment|'*=' |
+|op_DivisionAssignment|'/=' |
+|op_ModulusAssignment|'%=' |
+|op_BitwiseAndAssignment|'&=' |
+|op_BitwiseOrAssignment|'&#124;=' |
+|op_ExclusiveOrAssignment|'^=' |
+|op_LeftShiftAssignment|'<<='|
+|op_RightShiftAssignment| right_shift_assignment|
+|op_UnsignedRightShiftAssignment|unsigned_right_shift_assignment|
+
+However, it states that CLS compliance requires the operator methods to be non-void static methods with two parameters,
+i.e. matches what C# binary operators are. We should consider relaxing the CLS compliance requirements
+to allow the operators to be void returning instance methods with a single parameter.
+
+### Prefix increment and decrement operators
+
+See https://github.com/dotnet/csharpstandard/blob/draft-v8/standard/expressions.md#1296-prefix-increment-and-decrement-operators
+
+If `x` in `«op» x` is classified as a variable, then the priority is given to
+[instance increment operators](#increment-operators) as follows.
+
+First, an attempt is made to process the operation by applying
+[instance increment operator overload resolution](#instance-increment-operator-overload-resolution).
+If the process produces no result and no error, then the operation is processed
+by applying unary operator overload resolution as
+https://github.com/dotnet/csharpstandard/blob/draft-v8/standard/expressions.md#1296-prefix-increment-and-decrement-operators
+currently specifies.
+
+Otherwise, an operation `«op»x` is evaluated as follows.
+
+If type of `x` is known to be a reference type, the `x` is evaluated to get an instance `x₀`, the operator method is
+invoked on that instance, and `x₀` is returned as result of the operation.
+If `x₀` is `null`, the operator method invocation will throw a NullReferenceException.
+
+
+For example:
+``` C#
+var a = ++(new C()); // error: not a variable
+var b = ++a; // var temp = a; temp.op_Increment(); b = temp; 
+++b; // b.op_Increment();
+var d = ++C.P1; // error: setter is missing
+++C.P1; // error: setter is missing
+var e = ++C.P2; // var temp = C.op_Increment(C.get_P2()); C.set_P2(temp); e = temp;
+++C.P2; // var temp = C.op_Increment(C.get_P2()); C.set_P2(temp);
+
+class C
+{
+    public static C P1 { get; } = new C();
+    public static S P2 { get; set; } = new C();
+
+    public static C operator ++(C x) => ...;
+    public void operator ++() => ...;
+}
+```
+
+If type of `x` is not known to be a reference type:
+- If result of increment is used, the `x` is evaluated to get an instance `x₀`, the operator method is
+  invoked on that instance, `x₀` is assigned to `x` and `x₀` is returned as result of
+  the compound assignment.
+- Otherwise, the operator method is invoked on `x`.
```

Comment:
Should similar bullet points (if result is used or not) be above as well (when `x` is known to be a reference type)?

Reply:
> Should similar bullet points (if result is used or not) be above as well (when `x` is known to be a reference type)?

The difference is intentional